### PR TITLE
Update cluster-api, add cluster-name label to machines

### PR DESCRIPTION
**Pull request by vincepri**

### CODE REVIEW

**Comment by vincepri** [comment]
/assign @sflxn @frapposelli @figo 

Need some 👀on the changes, especially in Machine* labels to make sure they are correct.

**Comment by figo** [comment]
@vincepri thanks for the work, please update yaml for test as well at https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/master/scripts/e2e/bootstrap_job/spec/machines.template

we have yet to let test using example yaml.

**Comment by figo** [comment]
@vincepri  shouldn't cluster.yaml adding label as well? both the example yaml and test yaml template.

**Comment by figo** [comment]
> @vincepri shouldn't cluster.yaml adding label as well? both the example yaml and test yaml template.

found out, only the label need to match the cluster name in cluster.yaml

**Comment by sidharthsurana** [comment]
@vincepri Can you provide the motivation/need for this label? Also, the current set of changes are only going to result in these labels exist on the `Machine` objects. Was your intention to have these label show up in the underlying `Node` that these `Machine` create?

**Comment by figo** [comment]
> @vincepri Can you provide the motivation/need for this label? Also, the current set of changes are only going to result in these labels exist on the `Machine` objects. Was your intention to have these label show up in the underlying `Node` that these `Machine` create?

not trying to steal the conversation, just a quick answer, this is to support multi-cluster in one namespace, as discussed at issue: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/191

**Comment by sidharthsurana** [comment]
Thanks @figo I see the main reasoning being https://github.com/kubernetes-sigs/cluster-api/pull/728

File: vendor/sigs.k8s.io/cluster-api/pkg/controller/machine/controller.go [hunk 1]
```diff
@@ -194,18 +197,17 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
...
-		Namespace: machine.Namespace,
-		// This is set so the fake client can be used for unit test. See:
-		// https://github.com/kubernetes-sigs/controller-runtime/issues/168
-		Raw: &metav1.ListOptions{
-			TypeMeta: metav1.TypeMeta{
-				APIVersion: clusterv1.SchemeGroupVersion.String(),
-				Kind:       "Cluster",
-			},
-		},
+		FieldSelector: fields.OneTermEqualSelector("metadata.name", machine.Labels[MachineClusterLabelName]),
```
Comment (figo) [inline vendor/sigs.k8s.io/cluster-api/pkg/controller/machine/controller.go]: @vincepri, got the error: Index with name field:metadata.name does not exist.
it should be `fields.OneTermEqualSelector("metadata.name", machine.Labels[MachineClusterLabelName]).String()`?

**Comment by vincepri** [comment]
Waiting for https://github.com/kubernetes-sigs/cluster-api/pull/740

**Pull request merged**